ui/raidboss/data/06-ew/raid/p2n.ts

```diff
@@ -0,0 +1,178 @@
+import NetRegexes from '../../../../../resources/netregexes';
+import Outputs from '../../../../../resources/outputs';
+import { callOverlayHandler } from '../../../../../resources/overlay_plugin_api';
+import { Responses } from '../../../../../resources/responses';
+import ZoneId from '../../../../../resources/zone_id';
+import { RaidbossData } from '../../../../../types/data';
+import { PluginCombatantState } from '../../../../../types/event';
+import { TriggerSet } from '../../../../../types/trigger';
+
+export interface Data extends RaidbossData {
+  bodyActor?: PluginCombatantState;
+}
+
+const triggerSet: TriggerSet<Data> = {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  zoneId: ZoneId.AsphodelosTheSecondCircle,
+  timelineFile: 'p2n.txt',
+  triggers: [
+    {
+      id: 'P2N Murky Depths',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '680F', source: 'Hippokampos', capture: false }),
+      response: Responses.aoe(),
+    },
+    {
+      id: 'P2N Doubled Impact',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '680E', source: 'Hippokampos' }),
+      response: Responses.sharedTankBuster(),
+    },
+    {
+      id: 'P2N Spoken Cataract',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['67F8', '67F7', '67F9'], source: 'Hippokampos' }),
+      delaySeconds: 1,
+      promise: async (data) => {
+        const callData = await callOverlayHandler({
+          call: 'getCombatants',
+        });
+        if (!callData || !callData.combatants || !callData.combatants.length) {
+          console.error('SpokenCataract: failed to get combatants: ${JSON.stringify(callData)}');
+          return;
+        }
+        // This is the real hippo, according to hp.
+        const hippos = callData.combatants.filter((c) => c.BNpcID === 13721);
+        if (hippos.length !== 1) {
+          console.error('SpokenCataract: There is not exactly one Hippo?!?: ${JSON.stringify(hippos)}');
+          data.bodyActor = undefined;
+          return;
+        }
+        data.bodyActor = hippos[0];
+      },
+      alertText: (data, matches, output) => {
+        if (!data.bodyActor) {
+          console.error('SpokenCataract: No boss actor found. Did the promise fail?');
+          return;
+        }
+        const heading = data.bodyActor?.Heading;
+        let bodyHeading = null;
+
+        if (heading < -3.0 || heading > 3.0)
+          bodyHeading = 'north';
+        if (heading < -1.56 && heading > -1.58)
+          bodyHeading = 'west';
+        if (heading > -0.1 && heading < 0.1)
+          bodyHeading = 'south';
+        if (heading > 1.56 && heading < 1.58)
+          bodyHeading = 'east';
+
+        if (matches.id === '67F8') {
+          if (bodyHeading === 'north')
+            return output.nw!() + '/' + output.ne!();
```

Comment:
I should probably add this to the contributing/raidboss docs, but even though it's technically valid to return a string here, as a general rule I try not to do any string concatenation of outputs and instead use the output strings themselves.  For example, Hydaelyn uses a `combo` output string here: https://github.com/quisquous/cactbot/blob/main/ui/raidboss/data/06-ew/trial/hydaelyn-ex.ts#L73 that takes in the two parameters of the mechanics.  The reasoning here is that it then gives users full flexiblity to override the output in the UI.

Another alternative here is to say "North" and "North Intercards" (or "North Corners")? I find that a little bit easier to parse than NW/NE, but maybe you feel differently?

Reply:
I saw that line from hydaelin-ex actually but I have no idea how it works. Maybe my JavaScript fu is too weak, I'll try and look into it.

I guess intercards or corners would indeed be better but I just didn't think of it. I'll change it once I find out what "..." does and how these objects end up inside each other?!?

Reply:
`...` is "spread syntax": https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax.  I was just using it so that I could have a couple of different sets of outputstrings that built on top of each other.

I don't think the spread syntax is important here.  A better example than Hydaelyn is: https://github.com/quisquous/cactbot/blob/38e1c8ca5c6422be44bf831b38a86f9a1f5937c7/ui/raidboss/data/03-hw/trial/sophia-ex.ts#L416-L439 where you need one output string to be the combining version and can then pass in the parameters when you call `output.multiple({ dir1: etc, dir2: etc })`.

Reply:
Changed output to corner style in a8562bbece06d2fd3a33b1cbac9473feb52ee86b